### PR TITLE
feat: date-range filters for /articles/ and /trials/ endpoints

### PR DESCRIPTION
**Pull request by brunoamaral**
## Summary

- Adds `published_date_after` / `published_date_before` query params to `ArticleFilter` (filters `Articles.published_date`, a `DateTimeField`)
- Adds `date_registration_after` / `date_registration_before` query params to `TrialFilter` (filters `Trials.date_registration`, a `DateField`)
- No model or migration changes — both target fields already exist

## Why

The Advanced Search UI at `/search/` renders date-range inputs as disabled with the note "Date filtering pending API support." These params are the backend half; the frontend can now simply drop the `disabled` attribute.

## Implementation notes

- **`DateTimeField` inclusive-end semantics**: `lookup_expr="date__gte"` / `"date__lte"` applies the `__date` transform so that e.g. `published_date_before=2023-12-31` includes articles published at `2023-12-31 23:30`, not just at midnight.
- **`DateField`**: plain `gte` / `lte` suffices for `date_registration`.
- **400 on invalid input**: enforced automatically by `DjangoFilterBackend` (`raise_exception=True`) + `DateFilter`'s internal `DateField` validation. `input_formats=["%Y-%m-%d"]` rejects non-ISO formats.
- Filters are additive declarations — they AND with all existing params and don't touch pagination or CSV export.

## Tests (28 new)

`api/tests/test_date_range_filter_articles.py` and `api/tests/test_date_range_filter_trials.py`:
- after-only, before-only, closed range, neither (no-op)
- boundary inclusivity at both ends (including late-in-day edge case for `DateTimeField`)
- NULL `published_date` / `date_registration` rows are excluded when a filter is active
- Compose with `subjects`, `search`, `phase`
- `APIClient` tests: valid range → 200; invalid month (`2023-13-40`), non-date string, non-ISO format → 400

## Reviewer checklist

- [ ] `__date__gte` / `__date__lte` transform on `DateTimeField` is correct for the project's timezone setting
- [ ] No unintended interactions with `last_days` / `week` / `year` filters (they filter `discovery_date`, not `published_date`)

### CODE REVIEW

File: django/api/tests/test_date_range_filter_trials.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+			subject_name="Trial Date Subject",
+			subject_slug="trial-date-subject",
+			team=self.team,
+		)
+
+		def make_trial(title, reg_date):
+			t = Trials.objects.create(
+				title=title,
+				date_registration=reg_date,
+			)
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/api/tests/test_date_range_filter_trials.py:40]: Fixed — added explicit `link=f"https://example.com/trial-{n}"` to the `make_trial` helper and `link="https://example.com/trial-null"` to the null-date fixture.

File: django/api/tests/test_date_range_filter_trials.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+			t.subjects.add(self.subject)
+			return t
+
+		self.t_2018 = make_trial("Trial 2018-06-15", date(2018, 6, 15))
+		self.t_2019 = make_trial("Trial 2019-01-01", date(2019, 1, 1))
+		self.t_2021 = make_trial("Trial 2021-07-20", date(2021, 7, 20))
+		self.t_2022_end = make_trial("Trial 2022-12-31", date(2022, 12, 31))
+		self.t_2023 = make_trial("Trial 2023-03-10", date(2023, 3, 10))
+		# Trial with NULL date_registration
+		self.t_null = Trials.objects.create(title="Trial no date")
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/api/tests/test_date_range_filter_trials.py]: Fixed alongside the helper — the null-date trial now includes `link="https://example.com/trial-null"`.

File: django/api/filters.py [hunk 3]
```diff
@@ -95,6 +95,18 @@ class ArticleFilter(SubjectANDFilterMixin, filters.FilterSet):
...
+		label="Published date on or after (YYYY-MM-DD)",
+	)
+	published_date_before = filters.DateFilter(
+		field_name="published_date",
+		lookup_expr="date__lte",
+		input_formats=["%Y-%m-%d"],
+		label="Published date on or before (YYYY-MM-DD)",
+	)
 
 	class Meta:
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/api/filters.py:109]: Fixed — replaced `lookup_expr="date__gte/lte"` with method-based filters that build explicit datetime boundaries (`>= start-of-day`, `< start-of-next-day`), allowing PostgreSQL to use the btree index on `published_date`.

File: django/api/tests/test_date_range_filter_trials.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+				"date_registration_before": "2022-12-31",
+			},
+		)
+		self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+	def test_invalid_month_returns_400(self):
+		response = self.client.get(
+			"/trials/", {"date_registration_after": "2023-13-40"}
+		)
+		self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/api/tests/test_date_range_filter_trials.py:183]: No change needed. `DjangoFilterBackend` has `raise_exception = True` as a class-level default in django-filter, so invalid filter values already produce a 400. The 400 tests confirm this: all three pass in the container (`Bad Request: /trials/` lines visible in test output).

File: django/api/tests/test_date_range_filter_articles.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+			"/articles/",
+			{"published_date_after": "2023-01-01", "published_date_before": "2023-12-31"},
+		)
+		self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+	def test_invalid_month_returns_400(self):
+		response = self.client.get(
+			"/articles/", {"published_date_after": "2023-13-40"}
+		)
+		self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/api/tests/test_date_range_filter_articles.py:186]: No change needed — same reasoning as the /trials/ thread. `DjangoFilterBackend.raise_exception` defaults to `True`, so invalid dates already return 400. Confirmed by the passing `test_invalid_month_returns_400`, `test_non_date_string_returns_400`, and `test_non_iso_format_returns_400` cases.
